scripts/terraform_schema.py

```diff
@@ -0,0 +1,75 @@
+"""
+Save Terraform version and schema information to the file specified
+in AZUL_TRACKED_SCHEMA_PATH or verify that the Terraform information
+in that file is up-to-date.
+"""
+import argparse
+import json
+import logging
+from pathlib import (
+    Path,
+)
+import subprocess
+
+from azul import (
+    config,
+)
+from azul.deployment import (
+    terraform,
+)
+from azul.logging import (
+    configure_script_logging,
+)
+
+log = logging.getLogger(__name__)
+
+
+def run_terraform(*args: str) -> str:
+    terraform_dir = Path(config.project_root) / 'terraform'
+    cmd = subprocess.run(['terraform', *args],
+                         cwd=terraform_dir,
+                         check=True,
+                         stdout=subprocess.PIPE,
+                         text=True,
+                         shell=False)
+    return cmd.stdout
+
+
+def run_terraform_version() -> str:
+    # `terraform -version` prints a warning if you are not running the latest
+    # release of Terraform; we discard it, otherwise, we would need to update
+    # the tracked schema every time a new version of Terraform is released
+    versions, footer = run_terraform('-version').split('\n\n')
+    return versions
+
+
+def generate_schema_json() -> None:
+    schema = run_terraform('providers', 'schema', '-json')
+    tracked = {
```

Comment:
https://github.com/DataBiosphere/azul/pull/1701#discussion_r513836024

Please centralize reading and writing and parsing of the schema file in the Terraform class.